pkg/controller/agentregistration/server.go

```diff
@@ -74,9 +75,30 @@ func RunAgentRegistrationServer(ctx context.Context, port int, clientHolder *hel
 
 		// In the agent-registration case, the bootstrap sa is not created in the managed cluster namespace, because managed cluster is not created yet.
 		// Instead, it's in the pod namespace with the name "agent-registration-bootstrap".
-		bootstrapkubeconfig, _, err := bootstrap.CreateBootstrapKubeConfig(ctx, clientHolder, AgentRegistrationDefaultBootstrapSAName,
-			os.Getenv(constants.PodNamespaceEnvVarName),
-			7*24*3600, mergedKlusterletConfig)
+		ns := os.Getenv(constants.PodNamespaceEnvVarName)
+
+		var token []byte
+		if expirationStr == "" {
+			token, err = bootstrap.GetTokenFromSASecret(ctx, clientHolder.KubeClient, AgentRegistrationDefaultBootstrapSAName, ns)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+		} else {
+			expiration, err := time.ParseDuration(expirationStr)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+
+			token, _, err = bootstrap.RequestSAToken(ctx, clientHolder.KubeClient, AgentRegistrationDefaultBootstrapSAName, ns, int64(expiration.Seconds()))
```

Comment:
double check: will request a new token every time?

Reply:
Not every time, only if the `expiration` in the url parameter is set.
Regularly, if the `expiration` is not set, it will get token from SA secret.

Reply:
I am not sure about the case that the real expiration time returned by `bootstrap.RequestSAToken` is different from the expiration time the user requested(the case @zhiweiyin318 is working on), should we return the real expiration time to the user?

Reply:
Good point, I didn't look the PR https://github.com/stolostron/managedcluster-import-controller/pull/373 before, let me take a look now.

Reply:
The PR inspired me, the parameter name should be duration instead of expiration.

Reply:
Just like kubectl, it use duration in the `create token` command:

```
kubectl create token myapp --duration 10m
```

Reply:
Updated to `duration` in parameter.

Reply:
Echo the comment from @zhujian7 , the request expiration may not equals to the actual expiration returns by issuer:
```go
	// ExpirationSeconds is the requested duration of validity of the request. The
	// token issuer may return a token with a different validity duration so a
	// client needs to check the 'expiration' field in a response.
	// +optional
	ExpirationSeconds *int64 `json:"expirationSeconds" protobuf:"varint,4,opt,name=expirationSeconds"`
```